### PR TITLE
rb-generic-form has a 'hasRequiredFields' switch

### PR DESCRIPTION
**Pull request by jess0407**
- now can pass attr ‘has-required-fields’
- written the corresponding tests
- updated the component demo

https://rockabox.tpondemand.com/entity/10440

### CODE REVIEW

File: src/rb-generic-form/demo/demo.tpl.html [hunk 1]
```diff
@@ -21,5 +21,19 @@
                 </rb-generic-form>
             </div>
         </rb-section>
+        <rb-section heading="Has required fields.">
+            <div class="ComponentView">
+                <rb-generic-form has-required-fields='true'>
```
Comment (nwhite89) [inline src/rb-generic-form/demo/demo.tpl.html]: would you mind changing this to double quotes please same as the false, generally HTML should be written with double quotes but we haven't managed to integrate a decent HTML Linter as of yet.

**Comment by nwhite89** [comment]
other than that @ohskylab can you review look wise, the code is fine by me but if you could sign off it does what you need.

**Comment by ohskylab** [comment]
LGTM. Will need a squash. @jess0407, give me a shout if you want a hand with doing a fixup commit.

File: src/rb-generic-form/demo/demo.tpl.html (hunk 1 above)
Comment (jess0407) [inline src/rb-generic-form/demo/demo.tpl.html]: Noted and changed, thanks Nick.

**Pull request merged**